src/dataLayerManager.js

```diff
@@ -210,7 +218,11 @@ module.exports = function(config) {
         _dataLayer.splice(i, 1);
         i--;
       }
+
+      i++;
     }
+
+    _dataLayer.processed = true;
```

Comment:
I'd call the property `initialized` instead of `processed`. IMHO the data layer is `initialized` after all items have been `processed`.